.travis/update-types.ts

```diff
@@ -0,0 +1,316 @@
+  };
+  optional: boolean;
+}
+
+interface ArrayType {
+  type: 'array';
+  items: PrimitiveType | ObjectType | undefined;
+  optional: boolean;
+}
+
+const DOTS_REGEX = /\./g;
+const outFile = path.join(__dirname, '../types/index.d.ts');
+async function getLatestComments() {
+  const latestReleaseResponse = await got('https://api.github.com/repos/Palakis/obs-websocket/releases/latest', {json: true});
+  const latestReleaseTag = latestReleaseResponse.body.tag_name;
+  const commentsResponse = await got(`https://raw.githubusercontent.com/Palakis/obs-websocket/${latestReleaseTag}/docs/generated/comments.json`, {json: true});
+  return commentsResponse.body;
+}
+
+getLatestComments().then(rawComments => {
+  return parseApi(rawComments);
+}).catch(error => {
+  console.error(error);
+});
+
+function parseApi(raw: RawComments) {
+  const requestOverloads: string[] = [];
+  const eventOverloads: string[] = [];
+
+  Object.values(raw.requests).forEach(requestGroup => {
+    requestGroup.forEach(request => {
+      let requestTypeString = `send(requestType: '${request.name}'`;
+
+      if (request.params) {
+        requestTypeString += ', args: {';
+        const foo = unflattenAndResolveTypes(request.params);
+        requestTypeString += stringifyTypes(foo, {terminator: ',', finalTerminator: false});
+        requestTypeString += '}';
+      }
+
+      let returnTypeString = 'void';
+      if (request.returns) {
+        const foo = unflattenAndResolveTypes(request.returns);
+        returnTypeString = `{messageId: string;status: "ok";${stringifyTypes(foo, {terminator: ',', finalTerminator: false})}}`;
+        requestTypeString += `, callback?: (error: ObsError | null, data: ${returnTypeString}) => void`;
+      } else {
+        requestTypeString += ', callback?: (error: ObsError | null) => void';
+      }
+      requestTypeString += ')';
+
+      requestTypeString += `: Promise<${returnTypeString}>;`;
+      requestOverloads.push(requestTypeString);
+    });
+  });
+
+  Object.values(raw.events).forEach(eventGroup => {
+    eventGroup.forEach(event => {
+      let dataTypeString = 'data: {';
+      if (event.returns) {
+        const foo = unflattenAndResolveTypes(event.returns);
+        dataTypeString += stringifyTypes(foo);
+      }
+      dataTypeString += '}';
+
+      const eventType = `on(event: '${event.name}', listener: (${event.returns ? dataTypeString : ''}) => void): this;`;
+      eventOverloads.push(eventType);
+    });
+  });
+
+  /* tslint:disable:no-trailing-whitespace */
+  const sourceCode = `// This file is generated, do not edit.
+// TypeScript Version: 2.2
+/// <reference types="node" />
+import { EventEmitter } from 'events';
+
+export = ObsWebSocket;
+
+interface ObsError {
+  messageId: string;
+  status: "error";
+  error: string;
+}
+
+declare class ObsWebSocket extends EventEmitter {
+  connect(options?: {address?: string; password?: string}, callback?: (error?: Error) => void): Promise<void>;
```

Comment:
I guess it's most likely related to the lib itself rather than the TS bindings, but what is the point of having a callback along with a Promise?
Compatibility?

Reply:
Yeah, originally it was compatibility.
The old `obs-remote` supported callback functions so I had those available here.

I'll probably run through and update the docs to encourage using promises instead.

Reply:
I think it'd work better if you separate callback style and promise unless it's possible to use callback and promise at the same time